incubator/hnc/internal/reconcilers/anchor_test.go

```diff
@@ -50,11 +50,18 @@ var _ = Describe("Anchor", func() {
 		Eventually(getAnchorState(ctx, fooName, barName)).Should(Equal(api.Ok))
 	})
 
-	It("should set the anchor.status.state to Forbidden if the parent is not allowed to have subnamespaces", func() {
+	It("should remove the anchor in an excluded namespace", func() {
 		config.ExcludedNamespaces = map[string]bool{"kube-system": true}
 		kube_system_anchor_bar := newAnchor(barName, "kube-system")
 		updateAnchor(ctx, kube_system_anchor_bar)
-		Eventually(getAnchorState(ctx, "kube-system", barName)).Should(Equal(api.Forbidden))
+		Eventually(canGetAnchor(ctx, barName, "kube-system")).ShouldNot(Succeed())
+	})
+
+	It("should set the anchor.status.state to Forbidden if the subnamespace is an excluded namespace", func() {
+		config.ExcludedNamespaces = map[string]bool{"kube-system": true}
```

Comment:
Since `Ginkgo will normally run It blocks synchronously`, it's safer to have it set right above the test?

Reply:
I'm just wondering if we ever reset `config.ExcludedNamespaces` to its initial value between tests. Otherwise, the value we set in one test could persist to another test.

Maybe in the "before" block we could always reset it?

Reply:
Ah no it won't reset to initial value between tests, so the value set in one test could persist to another (if the `It` block is called before the other).

The other reason I prefer placing it in the `It` block is it's local/closer to the test case for readablitiy.

Reply:
_Setting_ the value in the `It` block sgtm. But maybe _clear_ it in the `BeforeEach` block so you know at the beginning of each test that it's starting in a known-good state?

Reply:
Maybe clear it in the `AfterEach`? Since in the `It` block, I set the value using a new map, not adding a new key/value pair to the existing map?

Reply:
Eventually, I decided to move the setting to `BeforeEach` from each `It` block, since `kube-system` is a well-know system(excluded) namespace and since the value is always set to `map[string]bool{"kube-system": true}`

Reply:
I actually do prefer it being explicit in the test. I'd prefer to see the following in `BeforeEach`:

`config.ExcludedNamespaces = nil`

That's all I'm going for...

Reply:
Done.